variables.tf

```diff
@@ -34,6 +34,12 @@ variable "log_analytics_workspace_enabled" {
   default     = true
 }
 
+variable "log_analytics_autoscaler_diagnostic_name" {
```

Comment:
Let's auto generate it based on the name of cluster-name. also let's pass the option of name_override